Formula/d/damask-grid.rb

```diff
@@ -0,0 +1,54 @@
+class DamaskGrid < Formula
+  desc "Grid solver of the Multi-physics simulation package - DAMASK"
+  homepage "https://damask-multiphysics.org"
+  url "https://damask-multiphysics.org/download/damask-3.0.0.tar.xz"
+  sha256 "aaebc65b3b10e6c313132ee97cfed427c115079b7e438cc0727c5207e159019f"
+  license "AGPL-3.0-only"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "fftw"
+  depends_on "hdf5-mpi"
+  depends_on "petsc"
+
+  uses_from_macos "zlib"
+
+  patch :p1, :DATA
```

Comment:
Is this submitted upstream? If not, could you do that and add a link to the submission?

Reply:
Here Martin from "upstream" ;)

DAMASK makes heavy use of PETSc. PETSc brings its own small package management system that allows to install or use existing libraries. If everything is configured as intended by PETSc, it provides the correct linker command. DAMASK relies on that. Unfortunately, the homebrew package (https://github.com/Homebrew/homebrew-core/blob/c37cd4c09e6bf4bfd3435f90a43fa97482117db6/Formula/p/petsc.rb) does not include external libraries. For some (hdf5-mpi, suite-sparse) it has a dependency, but makes no use of that (there is no `--with-hdf5=1` etc in the configure command), for others (notably FFTW for DAMASK grid), there is not even a dependency.

From upstream, there is no intention to fix the linker command: For manual installations, we simply ask people to configure PETSc with all needed dependencies and for the packages that we maintain we patch it. Supporting both cases (PETSc-specific or system wide) is unfortunately not easily possible.

What we could do (it's actually on my todo list) is to fix the PETSc homebrew package. For the packages, that are currently dependencies a "--with-XXX" makes absolutely sense. Otherwise, it would make sense to drop the dependency. But we should also discuss with the maintainers which dependencies should be included. For DAMASK, FFTW would be needed but there are more packages, e.g. boost. PETSc has the `--with-boost` option even thought it does not use it directly. But for packages relying on a correct linker command from PETSc, it would be quite handy.